src/hugo/content/en-us/patterns/MVC/_index.md

```diff
@@ -0,0 +1,88 @@
+---
+title: "Model-View-Controller Device Software Design"
+weight: 10
+aliases:
+  - /designs/MVC/
+summary: "recommendation for the design and structure of device software"
+---
+
+## Challenge
+
+IoT devices have a common set of concerns: 
+* report the current state of the device, attached sensors and actuators
+* processing control messages (both remote and local)
+* presenting status information 
+
+The design should also be easy to test and reuse across multiple projects, hardware platforms, languages, and SDKs. Some devices are 'one-way' only reporting state or taking action, while others are 'two-way,' performing both of these actions. The design should be generally composable to support all these uses as well as accommodate a flexible variety of peripherals, command sources, and interfaces. 
+
+## Solution
+
+An IoT device software design can be based on the [Model-View-Controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) design pattern.
+
+![Model-View-Controller Pattern](MVC-UML.png) 
```

Comment:
Does it make sense to add example actions that each component could do in a UML-type way? That would help identify where different actions would be targeted (e.g, change LED).

Reply:
i knew my _almost-uml_ would get me in trouble :P   
Added.